crates/host/src/secrets.rs

```diff
@@ -146,16 +153,19 @@ impl Manager {
             })
             // Retrieve the actual secret from the secrets backend
             .and_then(|secret_ref| async move {
+                let props = serde_json::from_str::<SecretReferenceProperties>(&secret_ref.policy_properties)
+                    .map_err(|_| anyhow::anyhow!("failed to deserialize secret reference properties"))?;
+                let properties = props.properties.get("backend").context("secret reference properties did not contain a backend key")?;
```

Comment:
just a naming suggestion:
```suggestion
                let backend = props.properties.get("backend").context("secret reference properties did not contain a backend key")?;
```